From bb98eab92d95b7315d27f4e59ae59b50587106d8 Mon Sep 17 00:00:00 2001
From: n-stein <16599929+n-stein@users.noreply.github.com>
Date: Thu, 2 May 2024 16:15:57 -0400
Subject: [PATCH] fix(#6693): workaround for bug ximion/appstream#631

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37474a5630..b3c2354e57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -708,6 +708,8 @@ if(LINUX)
                 COMMAND_ERROR_IS_FATAL ANY)
     endif()
 
+    execute_process(COMMAND sed -i -e "s/<developer_name>.*<\\/developer_name>//g" org.moneymanagerex.MMEX.metainfo.xml)
+
     # AppStream 'strict' validation wasn't an option prior to v0.15.4
     if(APPSTREAM_MAJOR_VERSION GREATER 0 OR (APPSTREAM_MINOR_VERSION GREATER 15 OR (APPSTREAM_MINOR_VERSION EQUAL 15 AND APPSTREAM_PATCH_VERSION GREATER_EQUAL 4)))
         execute_process(COMMAND appstreamcli validate --strict --no-net org.moneymanagerex.MMEX.metainfo.xml
