diff -Naur orig/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h patched/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h
--- orig/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h	2021-04-06 14:16:29.000000000 +0200
+++ patched/libdiscover/backends/FlatpakBackend/FlatpakTransactionThread.h	2021-04-06 18:12:57.312089675 +0200
@@ -7,11 +7,9 @@
 #ifndef FLATPAKTRANSACTIONTHREAD_H
 #define FLATPAKTRANSACTIONTHREAD_H
 
-extern "C" {
 #include <flatpak.h>
 #include <gio/gio.h>
 #include <glib.h>
-}
 
 #include <Transaction/Transaction.h>
 #include <QThread>
