diff -Naur orig/Source/WebKit/PlatformGTK.cmake patched/Source/WebKit/PlatformGTK.cmake
--- orig/Source/WebKit/PlatformGTK.cmake	2021-03-12 11:23:40.000000000 +0100
+++ patched/Source/WebKit/PlatformGTK.cmake	2021-04-03 00:24:38.118010508 +0200
@@ -684,7 +684,7 @@
             --nsversion=${WEBKITGTK_API_VERSION}
             --include=GObject-2.0
             --include=Gtk-${GTK_API_VERSION}.0
-            --include=Soup-2.4
+            --include=Soup-3.0
             --include-uninstalled=${CMAKE_BINARY_DIR}/JavaScriptCore-${WEBKITGTK_API_VERSION}.gir
             --library=webkit2gtk-${WEBKITGTK_API_VERSION}
             --library=javascriptcoregtk-${WEBKITGTK_API_VERSION}
@@ -693,7 +693,7 @@
             --no-libtool
             --pkg=gobject-2.0
             --pkg=${GTK_PKGCONFIG_PACKAGE}
-            --pkg=libsoup-2.4
+            --pkg=libsoup-3.0
             --pkg-export=webkit2gtk-${WEBKITGTK_API_VERSION}
             --output=${CMAKE_BINARY_DIR}/WebKit2-${WEBKITGTK_API_VERSION}.gir
             ${GIR_SOURCES_TOP_DIRS}
@@ -731,7 +731,7 @@
             --nsversion=${WEBKITGTK_API_VERSION}
             --include=GObject-2.0
             --include=Gtk-${GTK_API_VERSION}.0
-            --include=Soup-2.4
+            --include=Soup-3.0
             --include-uninstalled=${CMAKE_BINARY_DIR}/JavaScriptCore-${WEBKITGTK_API_VERSION}.gir
             --library=webkit2gtk-${WEBKITGTK_API_VERSION}
             --library=javascriptcoregtk-${WEBKITGTK_API_VERSION}
@@ -741,7 +741,7 @@
             --no-libtool
             --pkg=gobject-2.0
             --pkg=${GTK_PKGCONFIG_PACKAGE}
-            --pkg=libsoup-2.4
+            --pkg=libsoup-3.0
             --pkg-export=webkit2gtk-web-extension-${WEBKITGTK_API_VERSION}
             --output=${CMAKE_BINARY_DIR}/WebKit2WebExtension-${WEBKITGTK_API_VERSION}.gir
             ${GIR_SOURCES_TOP_DIRS}
