diff -Naur a/Makefile b/Makefile
--- a/Makefile	2023-10-14 13:52:11.000000000 +0200
+++ b/Makefile	2024-09-15 22:30:46.618197120 +0200
@@ -48,6 +48,7 @@
 DEFINES += $(shell xml2-config --cflags)
 DEFINES += -DSCRIPTFOLDER='"$(SKINDESIGNER_SCRIPTDIR)"'
 DEFINES += -D__STL_CONFIG_H
+DEFINES += -DDEPRECATED_CCONTROL=1
 
 INCLUDES += $(shell pkg-config --cflags freetype2 fontconfig)
 INCLUDES += $(shell pkg-config --cflags librsvg-2.0 cairo-png) -ljpeg
