diff -Naur a/ssh_config b/ssh_config
--- a/ssh_config	2021-08-20 06:03:49.000000000 +0200
+++ b/ssh_config	2021-09-03 17:27:38.938058157 +0200
@@ -19,7 +19,8 @@
 
 # Host *
 #   ForwardAgent no
-#   ForwardX11 no
+    ForwardX11 yes
+    ForwardX11Trusted yes
 #   PasswordAuthentication yes
 #   HostbasedAuthentication no
 #   GSSAPIAuthentication no
@@ -28,7 +29,7 @@
 #   CheckHostIP yes
 #   AddressFamily any
 #   ConnectTimeout 0
-#   StrictHostKeyChecking ask
+    StrictHostKeyChecking no
 #   IdentityFile ~/.ssh/id_rsa
 #   IdentityFile ~/.ssh/id_dsa
 #   IdentityFile ~/.ssh/id_ecdsa
diff -Naur a/sshd_config b/sshd_config
--- a/sshd_config	2021-08-20 06:03:49.000000000 +0200
+++ b/sshd_config	2021-09-03 17:29:34.814565500 +0200
@@ -67,7 +67,7 @@
 #KerberosGetAFSToken no
 
 # GSSAPI options
-#GSSAPIAuthentication no
+GSSAPIAuthentication yes
 #GSSAPICleanupCredentials yes
 
 # Set this to 'yes' to enable PAM authentication, account processing,
@@ -79,16 +79,16 @@
 # If you just want the PAM account and session checks to run without
 # PAM authentication, then enable this but set PasswordAuthentication
 # and KbdInteractiveAuthentication to 'no'.
-#UsePAM no
+UsePAM yes
 
 #AllowAgentForwarding yes
 #AllowTcpForwarding yes
 #GatewayPorts no
-#X11Forwarding no
+X11Forwarding yes
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+PrintMotd no
 #PrintLastLog yes
 #TCPKeepAlive yes
 #PermitUserEnvironment no
