diff -Naur orig/disk-utils/blockdev.c patched/disk-utils/blockdev.c
--- orig/disk-utils/blockdev.c	2021-05-06 12:55:46.000000000 +0200
+++ patched/disk-utils/blockdev.c	2021-06-09 21:02:45.095264977 +0200
@@ -234,6 +234,7 @@
 int main(int argc, char **argv)
 {
 	int fd, d, j, k;
+	int ndelay = 0;
 
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
@@ -263,6 +264,12 @@
 		return EXIT_SUCCESS;
 	}
 
+	if (!strcmp(argv[1], "-n")) {
+		ndelay = O_NDELAY;
+		argv++;
+		argc--;
+	}
+
 	/* do each of the commands on each of the devices */
 	/* devices start after last command */
 	for (d = 1; d < argc; d++) {
@@ -288,7 +295,7 @@
 	}
 
 	for (k = d; k < argc; k++) {
-		fd = open(argv[k], O_RDONLY, 0);
+		fd = open(argv[k], O_RDONLY | ndelay, 0);
 		if (fd < 0)
 			err(EXIT_FAILURE, _("cannot open %s"), argv[k]);
 		do_commands(fd, argv, d);
