From 1673c7961a89d1e3a2ba64cd48e3a9a2a32cf015 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Va=CC=81clav=20Slavi=CC=81k?= <vaclav@slavik.io>
Date: Mon, 2 Sep 2024 16:35:24 +0200
Subject: [PATCH] Compilation fix for !HAVE_HTTP_CLIENT

---
 src/errors.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/errors.cpp b/src/errors.cpp
index 295098b966..0fd4feda1a 100644
--- a/src/errors.cpp
+++ b/src/errors.cpp
@@ -27,7 +27,7 @@
 
 #include <wx/translation.h>
 
-#ifndef __WXOSX__
+#if defined(HAVE_HTTP_CLIENT) && !defined(__WXOSX__)
 #include <cpprest/http_client.h>
 #include <boost/algorithm/string.hpp>
 #endif
@@ -64,7 +64,7 @@ wxString errors::detail::DescribeExceptionImpl(Rethrower& rethrower)
     {
         return e.What();
     }
-#ifndef __WXOSX__
+#if defined(HAVE_HTTP_CLIENT) && !defined(__WXOSX__)
     catch (const web::http::http_exception & e)
     {
         // rephrase the errors more humanly; the default form is too cryptic
@@ -83,7 +83,7 @@ wxString errors::detail::DescribeExceptionImpl(Rethrower& rethrower)
         boost::trim_right(msg);
         return wxString::Format(_("Network error: %s (%d)"), from_c_string(msg.c_str()), e.error_code().value());
     }
-#endif // !__WXOSX__
+#endif // defined(HAVE_HTTP_CLIENT) && !defined(__WXOSX__)
     catch (const std::exception& e)
     {
         return from_c_string(e.what());
