diff -Naur orig/Makefile.am patched/Makefile.am
--- orig/Makefile.am	2017-08-17 02:38:43.000000000 +0200
+++ patched/Makefile.am	2021-03-18 23:21:00.036156152 +0100
@@ -27,7 +27,6 @@
              include \
              tools \
              clients \
-             doc \
-             demos
+             doc
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_AMFLAGS = -I .
diff -Naur orig/Makefile.in patched/Makefile.in
--- orig/Makefile.in	2017-12-05 13:49:57.000000000 +0100
+++ patched/Makefile.in	2021-03-18 23:21:34.572362419 +0100
@@ -390,8 +390,7 @@
              include \
              tools \
              clients \
-             doc \
-             demos
+             doc
 
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_AMFLAGS = -I .
