diff -Naur orig/source/graphics/ParticleManager.h patched/source/graphics/ParticleManager.h
--- orig/source/graphics/ParticleManager.h	2021-02-06 01:27:21.000000000 +0100
+++ patched/source/graphics/ParticleManager.h	2021-04-17 02:00:44.912118205 +0200
@@ -22,6 +22,7 @@
 #include "graphics/ParticleEmitterType.h"
 
 #include <boost/random/mersenne_twister.hpp>
+#include <list>
 #include <unordered_map>
 
 class SceneCollector;
