diff -Naur orig/conf.d/Makefile.am mod/conf.d/Makefile.am
--- orig/conf.d/Makefile.am	2019-10-16 12:05:28.000000000 +0200
+++ mod/conf.d/Makefile.am	2021-03-11 21:12:34.100896708 +0100
@@ -45,6 +45,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
diff -Naur orig/conf.d/Makefile.in mod/conf.d/Makefile.in
--- orig/conf.d/Makefile.in	2020-11-28 03:49:37.000000000 +0100
+++ mod/conf.d/Makefile.in	2021-03-11 21:12:34.100896708 +0100
@@ -398,6 +398,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
