From 977858f6e81f434bb82458b8524f25375b7ed605 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Wed, 10 Mar 2021 17:13:24 +0100
Subject: add ubk5


diff --git a/bash-completion/completions/build b/bash-completion/completions/build
index 6a5ec09..557504c 100644
--- a/bash-completion/completions/build
+++ b/bash-completion/completions/build
@@ -23,7 +23,7 @@ _buildubk3()
         foo="$(/bin/ls /usr/src/UHUBUILD/UB-UBK3)"
         COMPREPLY=($(compgen -W "$foo" -- "$cur"))
 }
-complete -F _buildubk2 build-ubk3
+complete -F _buildubk3 build-ubk3
 
 _buildubk4()
 {
@@ -32,4 +32,13 @@ _buildubk4()
         foo="$(/bin/ls /usr/src/UHUBUILD/UB-UBK4)"
         COMPREPLY=($(compgen -W "$foo" -- "$cur"))
 }
-complete -F _buildubk2 build-ubk4
+complete -F _buildubk4 build-ubk4
+
+_buildubk5()
+{
+        local cur foo
+        cur=${COMP_WORDS[COMP_CWORD]}
+        foo="$(/bin/ls /usr/src/UHUBUILD/UB-UBK5)"
+        COMPREPLY=($(compgen -W "$foo" -- "$cur"))
+}
+complete -F _buildubk5 build-ubk5
diff --git a/src/build/build b/src/build/build
index 4392f17..2577416 100755
--- a/src/build/build
+++ b/src/build/build
@@ -34,6 +34,8 @@ case `basename $0` in
 		export UHUBUILD_DISTR=UBK3;;
 	build-ubk4)
 		export UHUBUILD_DISTR=UBK4;;
+	build-ubk5)
+		export UHUBUILD_DISTR=UBK5;;
 esac
 
 UHUBUILD_DISTR_NUM=${UHUBUILD_DISTR/UBK/}
diff --git a/src/build/build-all b/src/build/build-all
index 42f792a..560136a 100755
--- a/src/build/build-all
+++ b/src/build/build-all
@@ -11,6 +11,6 @@
 	exit 1
 }
 
-for i in ubk1 ubk2 ubk3 ubk4; do
+for i in ubk3 ubk4 ubk5; do
 	/usr/bin/build-"$i" "$@"
 done
diff --git a/src/build/build-clean b/src/build/build-clean
index d97684b..283b52e 100755
--- a/src/build/build-clean
+++ b/src/build/build-clean
@@ -25,6 +25,8 @@ case ${BASENAME} in
 		export DISTR=UBK3;;
 	build-clean-ubk4)
 		export DISTR=UBK4;;
+	build-clean-ubk5)
+		export DISTR=UBK5;;
 esac
 
 [ "$UID" != 0 ] && {
diff --git a/src/build/build-clean-all b/src/build/build-clean-all
index f3817fb..c512794 100755
--- a/src/build/build-clean-all
+++ b/src/build/build-clean-all
@@ -9,6 +9,6 @@
 	exit 1
 }
 
-for i in ubk1 ubk2 ubk3 ubk4; do
+for i in ubk1 ubk2 ubk3 ubk4 ubk5; do
 	/usr/bin/build-clean-"$i" "$@"
 done
