diff -Naur orig/cmake/install_layout.cmake patched/cmake/install_layout.cmake
--- orig/cmake/install_layout.cmake	2021-02-19 10:41:01.000000000 +0100
+++ patched/cmake/install_layout.cmake	2021-03-11 23:42:19.496760559 +0100
@@ -182,24 +182,24 @@
 #
 SET(INSTALL_INCLUDEDIR_DEB              "include/mariadb")
 #
-SET(INSTALL_DOCDIR_DEB                  "share/doc")
-SET(INSTALL_DOCREADMEDIR_DEB            "share/doc")
+SET(INSTALL_DOCDIR_DEB                  "share/doc/mysql")
+SET(INSTALL_DOCREADMEDIR_DEB            "share/doc/mysql")
 SET(INSTALL_MANDIR_DEB                  "share/man")
 SET(INSTALL_INFODIR_DEB                 "share/info")
 #
 SET(INSTALL_SHAREDIR_DEB                "share")
 SET(INSTALL_MYSQLSHAREDIR_DEB           "share/mysql")
 SET(INSTALL_MYSQLTESTDIR_DEB            "share/mysql/mysql-test")
-SET(INSTALL_SQLBENCHDIR_DEB             ".")
+SET(INSTALL_SQLBENCHDIR_DEB             "")
 SET(INSTALL_SUPPORTFILESDIR_DEB         "share/mysql")
 #
 SET(INSTALL_MYSQLDATADIR_DEB            "/var/lib/mysql")
 
 SET(INSTALL_UNIX_ADDRDIR_DEB            "/run/mysqld/mysqld.sock")
-SET(INSTALL_SYSTEMD_UNITDIR_DEB         "/lib/systemd/system")
+SET(INSTALL_SYSTEMD_UNITDIR_DEB         "/usr/lib/systemd/system")
 SET(INSTALL_SYSTEMD_SYSUSERSDIR_DEB     "/usr/lib/sysusers.d")
 SET(INSTALL_SYSTEMD_TMPFILESDIR_DEB     "/usr/lib/tmpfiles.d")
-SET(INSTALL_PAMDIR_DEB                  "/lib/${CMAKE_CXX_LIBRARY_ARCHITECTURE}/security")
+SET(INSTALL_PAMDIR_DEB                  "/usr/lib/security")
 SET(INSTALL_PAMDATADIR_DEB              "/etc/security")
 
 #
diff -Naur orig/support-files/rpm/my.cnf patched/support-files/rpm/my.cnf
--- orig/support-files/rpm/my.cnf	2021-02-19 10:41:04.000000000 +0100
+++ patched/support-files/rpm/my.cnf	2021-03-11 23:41:08.548364607 +0100
@@ -7,5 +7,5 @@
 #
 # include *.cnf from the config directory
 #
-!includedir /etc/my.cnf.d
+!includedir /etc/mysql/conf.d
 
