diff -Naur orig/libmariadb/cmake/ConnectorName.cmake patched/libmariadb/cmake/ConnectorName.cmake
--- orig/libmariadb/cmake/ConnectorName.cmake	2021-02-19 10:41:02.000000000 +0100
+++ patched/libmariadb/cmake/ConnectorName.cmake	2021-03-29 20:44:31.957417743 +0200
@@ -22,7 +22,7 @@
     SET(MACHINE_NAME "x64")
   ELSE()
     SET(MACHINE_NAME "32")
-  END()
+  ENDIF()
 ENDIF()
 
 SET(product_name "mysql-connector-c-${CPACK_PACKAGE_VERSION}-${PLATFORM_NAME}${CONCAT_SIGN}${MACHINE_NAME}")
