diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2021-07-22 23:55:35.000000000 +0200
+++ patched/src/useradd.c	2021-07-27 14:50:09.293627853 +0200
@@ -101,8 +101,8 @@
 /*
  * These defaults are used if there is no defaults file.
  */
-static gid_t def_group = 1000;
-static const char *def_gname = "other";
+static gid_t def_group = 999;
+static const char *def_gname = "users";
 static const char *def_home = "/home";
 static const char *def_shell = "/bin/bash";
 static const char *def_template = SKEL_DIR;
