diff -Naur orig/src/useradd.c patched/src/useradd.c
--- orig/src/useradd.c	2021-08-02 23:12:21.019395160 +0200
+++ patched/src/useradd.c	2021-08-02 23:14:19.427707002 +0200
@@ -2378,7 +2378,7 @@
 	uid_t uid_min ;
 	uid_t uid_max ;
 	if(rflg){
-		uid_min = (uid_t)getdef_ulong("SYS_UID_MIN",101UL);
+		uid_min = (uid_t)getdef_ulong("SYS_UID_MIN",1UL);
 		uid_max = (uid_t)getdef_ulong("SYS_UID_MAX",getdef_ulong("UID_MIN",1000UL)-1);
 		if(uid_min <= uid_max){
 			if(user_id < uid_min || user_id >uid_max)
