diff -Naur orig/third_party/closure_compiler/compiler.py patched/third_party/closure_compiler/compiler.py
--- orig/third_party/closure_compiler/compiler.py	2021-06-06 19:40:05.954174642 +0200
+++ patched/third_party/closure_compiler/compiler.py	2021-06-06 19:58:55.097449124 +0200
@@ -12,9 +12,10 @@
 import subprocess
 
 
-_CURRENT_DIR = os.path.join(os.path.dirname(__file__))
-_JAVA_PATH = os.path.join(_CURRENT_DIR, "..", "jdk", "current", "bin", "java")
-assert os.path.isfile(_JAVA_PATH), "java only allowed in android builds"
+_CURRENT_DIR = "/usr"
+#_JAVA_PATH = os.path.join(_CURRENT_DIR, "lib", "java", "jre", "bin", "java")
+_JAVA_PATH = os.path.join(_CURRENT_DIR, "lib", "jvm", "openjdk-15", "bin", "java")
+# assert os.path.isfile(_JAVA_PATH), "java only allowed in android builds"
 
 class Compiler(object):
   """Runs the Closure compiler on given source files to typecheck them
