diff -Naur orig/apt-pkg/deb/dpkgpm.cc patched/apt-pkg/deb/dpkgpm.cc
--- orig/apt-pkg/deb/dpkgpm.cc	2021-04-13 17:26:59.000000000 +0200
+++ patched/apt-pkg/deb/dpkgpm.cc	2021-04-18 21:04:00.787244233 +0200
@@ -1875,8 +1875,8 @@
 	       if (I->File[0] != '/')
 		  return _error->Error("Internal Error, Pathname to install is not absolute '%s'",I->File.c_str());
 	       auto file = flNotDir(I->File);
-	       if (flExtension(file) != "deb")
-		  file.append(".deb");
+//	       if (flExtension(file) != "deb")
+//		  file.append(".deb");
 	       std::string linkpath;
 	       if (dpkg_recursive_install_numbered)
 		  strprintf(linkpath, "%s/%.*lu-%s", tmpdir_for_dpkg_recursive.get(), p, n, file.c_str());
