diff -Naur a/rules.d/60-persistent-storage.rules.in b/rules.d/60-persistent-storage.rules.in
--- a/rules.d/60-persistent-storage.rules.in	2024-08-15 22:46:49.000000000 +0200
+++ b/rules.d/60-persistent-storage.rules.in	2024-08-30 22:42:16.046900043 +0200
@@ -151,6 +151,9 @@
 ENV{ID_PATH}=="", GOTO="persistent_storage_by-path_parts_end"
 ENV{DEVTYPE}!="partition", GOTO="persistent_storage_by-path_parts_end"
 
+# by-type links (filesystem metadata)
+ENV{ID_FS_USAGE}=="filesystem|other|raid|crypto", ENV{ID_FS_TYPE}=="?*", SYMLINK+="disk/by-type/$env{ID_FS_TYPE}/%k"
+
 SYMLINK+="disk/by-path/$env{ID_PATH}-part/by-partnum/%n"
 ENV{ID_PART_ENTRY_UUID}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}-part/by-partuuid/$env{ID_PART_ENTRY_UUID}"
 ENV{ID_PART_ENTRY_SCHEME}=="gpt", ENV{ID_PART_ENTRY_NAME}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}-part/by-partlabel/$env{ID_PART_ENTRY_NAME}"
