diff -Naur a/src/journal/journald.conf b/src/journal/journald.conf
--- a/src/journal/journald.conf	2024-08-15 22:46:49.000000000 +0200
+++ b/src/journal/journald.conf	2024-08-30 22:52:02.969774757 +0200
@@ -24,17 +24,17 @@
 #SyncIntervalSec=5m
 #RateLimitIntervalSec=30s
 #RateLimitBurst=10000
-#SystemMaxUse=
+SystemMaxUse=50M
 #SystemKeepFree=
-#SystemMaxFileSize=
+SystemMaxFileSize=50M
 #SystemMaxFiles=100
-#RuntimeMaxUse=
+RuntimeMaxUse=50M
 #RuntimeKeepFree=
-#RuntimeMaxFileSize=
+RuntimeMaxFileSize=50M
 #RuntimeMaxFiles=100
 #MaxRetentionSec=0
 #MaxFileSec=1month
-#ForwardToSyslog=no
+ForwardToSyslog=yes
 #ForwardToKMsg=no
 #ForwardToConsole=no
 #ForwardToWall=yes
