diff -Naur orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	2021-06-30 15:10:00.000000000 +0200
+++ patched/CMakeLists.txt	2021-07-03 00:44:59.084334908 +0200
@@ -189,8 +189,8 @@
   set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/lib")
   set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/lib")
 
-  install(FILES README.txt README-commercial.txt kdsoap.pri kdwsdl2cpp.pri DESTINATION ${INSTALL_DOC_DIR})
-  install(DIRECTORY LICENSES DESTINATION ${INSTALL_DOC_DIR})
+  # install(FILES README.txt README-commercial.txt kdsoap.pri kdwsdl2cpp.pri DESTINATION ${INSTALL_DOC_DIR})
+  # install(DIRECTORY LICENSES DESTINATION ${INSTALL_DOC_DIR})
 
   # Generate library version files
   include(ECMSetupVersion)
